scylla/src/frame/mod.rs

```diff
@@ -17,21 +17,69 @@ mod cql_types_test;
 use crate::frame::frame_errors::FrameError;
 use crate::transport::Compression;
 use bytes::{Buf, BufMut, Bytes};
-use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+use tokio::io::{AsyncRead, AsyncReadExt};
 use uuid::Uuid;
 
 use std::convert::TryFrom;
 
 use compress::lz4;
-use request::RequestOpcode;
+use request::Request;
 use response::ResponseOpcode;
 
+const HEADER_SIZE: usize = 9;
+
 // Frame flags
 pub const FLAG_COMPRESSION: u8 = 0x01;
 pub const FLAG_TRACING: u8 = 0x02;
 pub const FLAG_CUSTOM_PAYLOAD: u8 = 0x04;
 pub const FLAG_WARNING: u8 = 0x08;
 
+pub struct SerializedRequest {
+    data: Vec<u8>,
+}
+
+// TODO: Should we support serialization in other versions of the protocol than V4?
```

Comment:
There's probably no reason to support older versions, but there's already a v5 of the CQL binary protocol, which we eventually do also want to support.

Reply:
OK. I'll remove this comment.

In the future, it should be very easy to add support for v5 here; I looked at the protocol specs and the format of the header is exactly the same as in v4 (and v3), you only need to put a different number in the "version" field.